services/shhext/mailrequests.go

```diff
@@ -0,0 +1,130 @@
+package shhext
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	whisper "github.com/status-im/whisper/whisperv6"
+)
+
+// EnvelopeState in local tracker
+type EnvelopeState int
+
+const (
+	// NotRegistered returned if asked hash wasn't registered in the tracker.
+	NotRegistered EnvelopeState = -1
+	// EnvelopePosted is set when envelope was added to a local whisper queue.
+	EnvelopePosted EnvelopeState = iota
+	// EnvelopeSent is set when envelope is sent to atleast one peer.
+	EnvelopeSent
+	// MailServerRequestSent is set when p2p request is sent to the mailserver
+	MailServerRequestSent
+)
+
+// MailRequestMonitor responsible for monitoring history request to mailservers.
+type MailRequestMonitor struct {
```

Comment:
It's quite similar to `EnvelopesMonitor`. I wonder if we can unify both but that might be a next PR.

Reply:
the whole point was to split them because i am heavily changing EnvelopesMonitor.